chips/x86_q35/src/chip.rs

```diff
@@ -88,10 +109,24 @@ impl<'a, const PR: u16> Chip for Pc<'a, PR> {
                         self.com1.handle_interrupt();
                         self.com3.handle_interrupt();
                     }
-                    _ => unimplemented!("interrupt {num}"),
+                    _ => {
+                        // Convert back to physical interrupt line number before passing to
+                        // board-specific handler
+                        let phys_num = num - PIC1_OFFSET as u32;
+                        handled = unsafe { self.int_svc.service_interrupt(phys_num) };
+                    }
                 }
 
                 poller.clear_pending(num);
+
+                // Unmask the interrupt so it can fire again, but only if we know how to handle it
+                if handled {
+                    unsafe {
+                        crate::pic::unmask(num);
+                    }
+                } else {
+                    kernel::debug!("Unhandled external interrupt {} left masked", num);
```

Comment:
Is it possible that this will flood the console?

Reply:
I can find examples of both... nrf52 [panics](https://github.com/tock/tock/blob/2eafe5cc1b13bf1f0936b3bb7fc976f00c0aef2c/chips/nrf52/src/chip.rs#L124C21-L124C65), and e310x [prints a message](https://github.com/tock/tock/blob/2eafe5cc1b13bf1f0936b3bb7fc976f00c0aef2c/chips/e310x/src/chip.rs#L106). I have no preference at all, so if we think a panic is the correct/predominant approach I'm happy to make that change right away

Reply:
Given that angle, I think the current implementation (printing a message and leaving the interrupt masked) would be the better option for this chip. Partly because the chip probably hasn't received the same level of scrutiny as nrf52, and partly because x86 feels like a bit more "modular" of a platform anyways, where an unexpected interrupt from an unexpected device maybe isn't as notable/panic-worthy.